R/class-yspec.R

```diff
@@ -424,26 +431,32 @@ yspec_yml_file.default <- function(x,...) {
 #' 
 #' data <- ys_help$data()
 #' 
-#' data <- ys_add_labels(data,spec)
+#' data <- ys_add_labels(data, spec)
 #' 
-#' sapply(data,attr,"label")
+#' sapply(data, attr, "label")
 #' 
 #' str(data[,1:5])
 #' 
 #' @md
 #' @export
-ys_add_labels <- function(data,spec,fun=label.ycol) {
+ys_add_labels <- function(data, spec, fun = label.ycol, strict = TRUE) {
   assert_that(inherits(data,"data.frame"))
   assert_that(inherits(spec,"yspec"))
-  assert_that(identical(names(data),names(spec)))
-  col_labels <- map_chr(spec,fun)
-  for(i in seq_along(data)) {
-    attr(data[[i]],"label") <- col_labels[[i]]
+  col_labels <- map_chr(spec, fun)
```

Comment:
nit-pick of no consequence: since `col_labels` isn't used until after the guard on 446-448, to my eyes it'd be nicer to move it below so it's the line before `col_labels <- col_labels[names(col_labels) %in% names(data)]`.

Reply:
IC; yep, totally agree ... I like doing this in the code. Thanks for picking it up.